src/fingerprint-generator.ts

```diff
@@ -1,28 +1,51 @@
-const { default: ow } = require('ow');
-const HeaderGenerator = require('header-generator');
+import { HeaderGenerator, HeaderGeneratorOptions } from 'header-generator';
 
-const { BayesianNetwork } = require('generative-bayesian-network');
+// @ts-expect-error No ts definition
+import { BayesianNetwork } from 'generative-bayesian-network';
 
-const fingerprintNetworkDefinition = require('./data_files/fingerprint-network-definition.json');
+import fingerprintNetworkDefinition from './data_files/fingerprint-network-definition.json';
+import { MISSING_VALUE_DATASET_TOKEN, STRINGIFIED_PREFIX } from './constants';
 
-const STRINGIFIED_PREFIX = '*STRINGIFIED*';
-const MISSING_VALUE_DATASET_TOKEN = '*MISSING_VALUE*';
+export type ScreenFingerprint = {
+    availHeight: number,
+    availWidth: number,
+    pixelDepth: number,
+    height: number,
+    width: number,
+}
+
+export type NavigatorFingerprint = {
+    cookieEnabled: boolean,
+    doNotTrack: string,
+    language: string,
+    languages: string[],
+    platform: string,
+    deviceMemory?: number, // Firefox does not have deviceMemory available
+    hardwareConcurrency: number,
+    productSub: string,
+    vendor: string,
+    maxTouchPoints?: number;
+}
 
-const browserSpecificationShape = {
-    name: ow.string,
-    minVersion: ow.optional.number,
-    maxVersion: ow.optional.number,
-    httpVersion: ow.optional.string,
-};
+type WebGl = {
+    vendor: string,
+    renderer: string,
+}
 
-const headerGeneratorOptionsShape = {
-    browsers: ow.optional.array.ofType(ow.any(ow.object.exactShape(browserSpecificationShape), ow.string)),
-    operatingSystems: ow.optional.array.ofType(ow.string),
-    devices: ow.optional.array.ofType(ow.string),
-    locales: ow.optional.array.ofType(ow.string),
-    httpVersion: ow.optional.string,
-};
+export type Fingerprint = {
+    screen: ScreenFingerprint,
+    navigator: NavigatorFingerprint,
+    webGl: WebGl,
+    userAgent: string,
+    videoCodecs: Record<string, string>,
+    audioCodecs: Record<string, string>,
+    battery?: boolean,
+}
 
+type GetFingerprintReturnType = {
```

Comment:
Maybe this could be named differently? Something like "Credentials" or something similar in meaning. This seems clunky :)